pallets/gear-builtin/src/staking.rs

```diff
@@ -122,18 +122,19 @@ where
                 * 11
                 / 10;
         if payload.len() > max_payload_size {
-            return (Err(BuiltinActorError::DecodingError), 0);
+            return Err(BuiltinActorError::DecodingError);
         }
 
         // Decode the message payload to derive the desired action
-        let Ok(request) = Request::decode(&mut payload) else {
-            return (Err(BuiltinActorError::DecodingError), 0);
-        };
+        let request =
+            Request::decode(&mut payload).map_err(|_| BuiltinActorError::DecodingError)?;
 
         // Handle staking requests
         let call = Self::cast(request);
-        let (result, gas_spent) = Pallet::<T>::dispatch_call(origin, call, gas_limit);
+        Pallet::<T>::dispatch_call(origin, call, context).map(|_| Default::default())
```

Comment:
> why is it always returns default also?

The `handle` signature assumes it always returns some payload in case of success, even if no meaningful return message is expected by the sender. This is exactly the case for the `staking` builtin who just relays messages converting them into dispatchable calls for the `pallet_staking`.